lua/indentmini/init.lua

```diff
@@ -46,11 +42,32 @@ local function indentline()
       indent = prev > 20 and 4 or prev
     end
 
-    local hi_name = hl_group()
-
     ctx[row] = indent
 
-    for i = 1, indent - 1, indent_step(bufnr) do
+    local shiftw = indent_step(bufnr)
+    local last_defined_level = 0
+    for i = 1, indent - 1, shiftw do
+      local indent_level = math.floor((i - 1) / shiftw) + 1
+      local hi_name = ('IndentLine%d'):format(indent_level)
+      -- Attempt to fetch highlight details for the current level
+      local hl_details = api.nvim_get_hl(0, { name = hi_name })
+
+      -- If no custom highlight details are found, use the last defined level for looping back
+      if next(hl_details) == nil then
```

Comment:
if no custom highlight can we still just use `IndentLine` ?

Reply:
Yes, but not both. If you have `IndentLineX`, `IndentLine` is ignored.

1. All indents the same color (same as now)
```
vim.api.nvim_set_hl(0, 'IndentLine', { fg = '#794e17' })
-- OR
-- vim.api.nvim_set_hl(0, 'IndentLine1', { fg = '#794e17' })
```
2. Indent color rotates 1 -> 2 -> 3 -> 1 -> 2 -> 3 ...
```
vim.api.nvim_set_hl(0, 'IndentLine1', { fg = '#794e17' })
vim.api.nvim_set_hl(0, 'IndentLine2', { fg = '#4e4c99' })
vim.api.nvim_set_hl(0, 'IndentLine3', { fg = '#506003' })
```

3. Same as 2, `IndentLine` is ignored.
```
vim.api.nvim_set_hl(0, 'IndentLine', { fg = '#794e17' })
vim.api.nvim_set_hl(0, 'IndentLine1', { fg = '#794e17' })
vim.api.nvim_set_hl(0, 'IndentLine2', { fg = '#4e4c99' })
vim.api.nvim_set_hl(0, 'IndentLine3', { fg = '#506003' })
```